### ISSUE TITLE
[TRACE SDK] Batch Span Processor options not using environment variables

### ISSUE THREAD

**Issue by marcalff** [open-telemetry/opentelemetry-cpp#2085]
In this code:

```
struct BatchSpanProcessorOptions
{
  /**
   * The maximum buffer/queue size. After the size is reached, spans are
   * dropped.
   */
  size_t max_queue_size = 2048;

  /* The time interval between two consecutive exports. */
  std::chrono::milliseconds schedule_delay_millis = std::chrono::milliseconds(5000);

  /**
   * The maximum batch size of every export. It must be smaller or
   * equal to max_queue_size.
   */
  size_t max_export_batch_size = 512;
};
```

the default values for each configuration option are hard coded.

The [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#batch-span-processor) requires to use the following environment variables:

* `OTEL_BSP_SCHEDULE_DELAY`
* `OTEL_BSP_EXPORT_TIMEOUT`
* `OTEL_BSP_MAX_QUEUE_SIZE`
* `OTEL_BSP_MAX_EXPORT_BATCH_SIZE`

Note that the export timeout option is also missing from `BatchSpanProcessorOptions`.

**Comment by oandreeva-nv** [comment]
Is there any estimation on when this will be added?

**Comment by nikhilbhatia08** [comment]
Hi @marcalff I would like to work on this issue, We could use `opentelemetry::sdk::common::GetStringEnvironmentVariable()` here and then set them in `BatchProcessorOptions`,Thanks!

**Comment by marcalff** [comment]
@nikhilbhatia08 

If you feel like it, there is also `BatchLogRecordProcessorOptions`, for a separate PR.

**Comment by nikhilbhatia08** [comment]
@marcalff Yes, then the variables in `BatchLogRecordProcessorOptions` could also be initialized with environment variables right ? and the `export_timeout` could also be added ?
